backend/routes/table.js

```diff
@@ -0,0 +1,97 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+
+import connection from '../database';
+
+const router = express.Router();
+
+router.use(bodyParser.urlencoded({ extended: true }));
+
+/**
+ * @swagger
+ *
+ * /table/free:
+ *   get:
+ *     description: Get a list of of hours where there is a table free for a restaurent
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - name: restaurantID
+ *         description: Primary Key of Restaurant database table
+ *         in: query
+ *         required: true
+ *         type: string
+ *       - name: numberOfGuests
+ *         description: The number of guests that the table is being booked for
+ *         in: query
+ *         required: true
+ *         type: integer
+ *       - name: date
+ *         description: Date for when the booking is made
+ *         in: query
+ *         required: true
+ *         type: string
+ *     responses:
+ *        200:
+ *         description: Returns the list of times that the restaurent has a table free
+ */
+router.get('/free', async (req, res) => {
+  const { date, numberOfGuests, restaurantID } = req.query;
+  // Extracting day of the week from date
+  const reservationDate = new Date(date);
+  const dayNames = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+  const dayOfWeekIndex = reservationDate.getDay();
+  const day = dayNames[dayOfWeekIndex];
+
+
+  // Getting opening hours for the restaurant for that day
+  const { error: queryError, result: restaurentHours } = await connection.asyncQuery(
+    'SELECT OpenTime, CloseTime from HOURS WHERE RestaurantID = ? AND DayOfWeek = ?', [restaurantID, day]
+  );
+
+  if (queryError) {
+    res.status(500).json({ error: 'Internal server error' });
+    return;
+  }
+
+  if (!restaurentHours[0]) {
```

Comment:
You may want to check if restaurentHours is an array before indexing, eg
if (!Array.isArray(restaurentHours) || !restaurentHours[0]) {